lib/jitter/sleep.go

```diff
@@ -15,11 +19,10 @@ func Jitter(baseMs, maxMs, attempts int) time.Duration {
 		return time.Duration(0)
 	}
 
-	// Check for overflows when computing base * 2 ** attempts.
-	if attemptsMaxMs := baseMs * (1 << attempts); attemptsMaxMs > 0 {
+	// Cap the attempts to be 30 so we don't have integer overflows.
```

Comment:
- This would need to be `min` instead of `max`, otherwise we're going to use 30 as attempts when attempts is 1.
- Capping `attempts` doesn't change the behavior of this code as it already handles the overflows just fine: overflows will cause `attemptsMaxMs` to be negative and we'll use `maxMs` instead in that case). 
- if `baseMs` is >= `8589934592` `2** 30 * 8589934591` will still overflow on a 64-bit machine (and `baseMs` >= 2 will overflow on a 32-bit machine)
- If we want to explicitly check for overflows rather than rely on them producing negative numbers then we should do something like:
```
if x := powerOfTwo(min(attempts, 30)); math.MaxInt/baseMs <= x {
    // No overflow.
    maxMs = min(maxMs, baseMs*x)
}
```

Reply:
Oops, yeah it should be min